### PR TITLE
refactor: Refactor value and faucet integration tests

### PR DESCRIPTION
**Pull request by jkrvivian**
# Description of change

Part of #1187 
* Remove TestFaucetPersistence test
* Rename `faucetmoveallfunds_test.go ` to `preparefaucet_test.go`
* Combine `TestTransactionsPersistence` and `TestValueColoredPersistence`
* Add `AwaitPeerGetFundsFromFaucet`
* Fixes on other `Await...` functions
* Replace most of the sleeps with `Await...` functions

## Type of change

- Bug fix 
- Enhancement 

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tools/integration-tests/tester/tests/testutil.go [hunk 1]
```diff
@@ -187,7 +192,7 @@ func AwaitMessageAvailability(peers []*framework.Peer, messageIDs map[string]Dat
 	var missingMu sync.RWMutex
 	missing = map[identity.ID]map[string]types.Empty{}
 
-	for i := 0; time.Since(s) < waitFor; time.Sleep(500 * time.Millisecond) {
+	for i := 0; time.Since(s) < waitFor; time.Sleep(retryInterval) {
```
Comment (GalRogozinski) [inline tools/integration-tests/tester/tests/testutil.go:195]: :+1:

File: tools/integration-tests/tester/tests/testutil.go [hunk 2]
```diff
@@ -703,19 +780,36 @@ func AwaitTransactionInclusionState(peers []*framework.Peer, transactionIDs map[
...
+		i++
 	}
 	return ErrTransactionStateNotSameInTime
 }
 
+// AwaitCheck waits until the given peers are in synced.
+func AwaitCheck(f func(t *testing.T, peers []*framework.Peer), maxAwait time.Duration) error {
+	s := time.Now()
+	for ; time.Since(s) < maxAwait; time.Sleep(retryInterval) {
+
```
Comment (GalRogozinski) [inline tools/integration-tests/tester/tests/testutil.go:808]: isn't there some polling logic missing here?

File: tools/integration-tests/tester/tests/testutil.go [hunk 3]
```diff
@@ -664,13 +722,32 @@ func AwaitTransactionAvailability(peers []*framework.Peer, transactionIDs []stri
 // the the transactions exist beforehand.
 func AwaitTransactionInclusionState(peers []*framework.Peer, transactionIDs map[string]ExpectedInclusionState, maxAwait time.Duration) error {
 	s := time.Now()
-	for ; time.Since(s) < maxAwait; time.Sleep(1 * time.Second) {
+	var unmatchedMu sync.Mutex
+	unmatched := make(map[identity.ID]map[string]types.Empty)
+	// the element will be removod if a peer has the message with the given inclusion state
```
Comment (GalRogozinski) [inline tools/integration-tests/tester/tests/testutil.go:727]:
```suggestion
	// the element will be removed if a peer has the message with the given inclusion state
```

File: tools/integration-tests/tester/tests/testutil.go [hunk 4]
```diff
@@ -664,13 +722,32 @@ func AwaitTransactionAvailability(peers []*framework.Peer, transactionIDs []stri
 // the the transactions exist beforehand.
 func AwaitTransactionInclusionState(peers []*framework.Peer, transactionIDs map[string]ExpectedInclusionState, maxAwait time.Duration) error {
```
Comment (GalRogozinski) [inline tools/integration-tests/tester/tests/testutil.go:723]: since this function is a bit long, can you please either break it up or add some comments?

**Review by GalRogozinski — CHANGES_REQUESTED** [review]
Looks good, but I am mainly asking about the `AwaitCheck` function`

File: tools/integration-tests/tester/tests/testutil.go (hunk 2 above)
Comment (jkrvivian) [reply tools/integration-tests/tester/tests/testutil.go:808]: I think I messed things up during merging files. :stuck_out_tongue: 
We don't need this function.

File: tools/integration-tests/tester/tests/testutil.go (hunk 4 above)
Comment (jkrvivian) [reply tools/integration-tests/tester/tests/testutil.go:723]: I broke check inclusion state to another function.

**Comment by Wollac** [comment]
The #1478 contains a lot of changes also in the 'testutil.go'. So I've merge the changes from here into there.

**Pull request closed without merging**